src/exercises/exerciseTypes/multipleChoiceContext/MultipleChoiceContext.js

```diff
@@ -0,0 +1,157 @@
+import { useState, useEffect, useContext } from "react";
+import * as s from "../Exercise.sc.js";
+import strings from "../../../i18n/definitions";
+import NextNavigation from "../NextNavigation";
+import LoadingAnimation from "../../../components/LoadingAnimation.js";
+import InteractiveText from "../../../reader/InteractiveText.js";
+import { SpeechContext } from "../../../contexts/SpeechContext.js";
+import useSubSessionTimer from "../../../hooks/useSubSessionTimer.js";
+import shuffle from "../../../assorted/fisherYatesShuffle";
+
+const EXERCISE_TYPE = "Select_L2T_fitting_L2W";
+export default function MultipleChoiceContext({
+  api,
+  bookmarksToStudy,
+  correctAnswer,
+  notifyIncorrectAnswer,
+  setExerciseType,
+  isCorrect,
+  setIsCorrect,
+  moveToNextExercise,
+  toggleShow,
+  reload,
+  setReload,
+  exerciseSessionId,
+  activeSessionDuration,
+}) {
+  const [messageToAPI, setMessageToAPI] = useState("");
+  const [exerciseBookmarks, setExerciseBookmarks] = useState(null);
+  const [articleInfo, setArticleInfo] = useState(null);
+  const [interactiveText, setInteractiveText] = useState(null);
+  const speech = useContext(SpeechContext);
+  const [getCurrentSubSessionDuration] = useSubSessionTimer(
+    activeSessionDuration,
+  );
+  const [incorrectAnswer, setIncorrectAnswer] = useState("");
+  const [clickedIndex, setClickedIndex] = useState(null);
+  const [clickedOption, setClickedOption] = useState(null);
+  const [showSolution, setShowSolution] = useState(false);
+
+  useEffect(() => {
+    setExerciseType(EXERCISE_TYPE);
+    let initExerciseBookmarks = [...bookmarksToStudy];
+    for (let i = 0; i < initExerciseBookmarks.length; i++) {
+      if (i === 0)
+        initExerciseBookmarks[i].isExercise = true
+      else
+        initExerciseBookmarks[i].isExercise = false
+    }
+    setExerciseBookmarks(initExerciseBookmarks);
+    api.getArticleInfo(bookmarksToStudy[0].article_id, (articleInfo) => {
+      setInteractiveText(
+        new InteractiveText(
+          bookmarksToStudy[0].context,
+          articleInfo,
+          api,
+          "TRANSLATE WORDS IN EXERCISE",
+          EXERCISE_TYPE,
+          speech,
+        ),
+      );
+    setArticleInfo(articleInfo);
+    });
+  }, []);
+
+  useEffect(() => {
+    if (articleInfo && interactiveText) {
+      setExerciseBookmarks(shuffle(
```

Comment:
Now that I look at this, I think shuffling the list at line 49 might be easier to read.